ui/src/api.ts

```diff
@@ -78,6 +78,57 @@ export async function deleteAnnotation(arxivId: string, id: number) {
   return await axios.delete(`/api/v0/papers/arxiv:${arxivId}/annotation/${id}`);
 }
 
+export async function addLibraryEntry(paperId: string, paperTitle: string) {
+  const s2ApiUrl = `https://www.semanticscholar.org/api/1/library/entries`;
+  const tags: string[] = [];
+  const response = await axios.post(
+      s2ApiUrl,
+      {
+        paperId,
+        paperTitle,
+        tags
+      },
+      {withCredentials: true}
+  );
+  return response.data;
+}
+
+export async function getUserLibraryInfo(userId?: number) {
+  if (localStorage && userId) {
+    const maybeItem = localStorage.getItem(userId.toString());
+    if (maybeItem) {
+      const parsedItem: CachedUserLibrary = JSON.parse(maybeItem);
+      if (parsedItem.expires > Date.now()) {
+        console.debug(`Using cached library entries for:${userId}`);
+        return Promise.resolve(parsedItem.userLibrary);
+      } else {
+        console.debug(`Library entries for ${userId} are cached but have expired.`);
+      }
+    }
+  }
+
+  const s2ApiUrl = `https://www.semanticscholar.org/api/1/user`;
+  const data = await doGet(axios.get<UserInfo>(s2ApiUrl, {withCredentials: true}));
```

Comment:
If I understand right, the S2 API doesn't take a user ID as a parameter, but instead probably does some magic with cookies or session variables to detect the user. Is that the right interpretation? And in that case, the user ID is just used as a key for the cache?

Reply:
Yes, I am using the userId as the cache key because I figured it was the most unique thing I could use.